src/router.rs

```diff
@@ -125,46 +126,101 @@ impl RequestForwarder {
         Ok(())
     }
 
-    #[allow(clippy::single_match_else)]
-    pub fn forward_request(&self, request: ComponentRequest) -> Result<Response<Body>, RouterError> {
-        let component_name = format!("{}/{}", request.user, request.repo);
-        //if !self.components_map.read().contains_key(&component_name) {
-        //    self.update_workers()?;
-        //}
-
-        // TODO: Make this more efficient (when do I actually need to call update_workers()?)
-        self.update_workers()?;
-
-        debug!("Forwarding request to: {:?}", component_name);
-        let locked_map = self.components_map.read();
-        let target_component_option = locked_map.get(&component_name);
-
-        let target_component = match target_component_option {
-            Some(v) => v,
-            None => {
-                let body = Body::from("Requested component not found\n");
-                warn!("Component to forward to not found: {:?}", component_name);
-                return Ok(Response::builder().status(404).body(body).unwrap());
-            }
-        };
+    fn find_appropriate_worker(&self, component_name: &str) -> Option<Arc<WorkerNode>> {
```

Comment:
Glad you added this function, will be useful when we actually start doing educated decisions about which worker to send a request to